### DISCUSSION TITLE
To confirm the convention of edge_index

### DISCUSSION THREAD

**Discussion by zcaicaros** [pyg-team/pytorch_geometric#2556] [Q&A] [answered]
Sorry for disturbing you guys with this stupid question. My algorithm does not work and I have been debugging for a long time. So I would like to confirm every bit of the implementation is as expected.

If I have a graph like below, two nodes A pointing to B
```
A -> B
```
which is directed acyclic graph (DAG), and I have a adjacency matrix for its connection, say
```
[[1, 1],
 [0, 1]]
```
the diagonals have value 1 meaning add self-loop for each node.

Next I want to create a `torch_geometric.data.Data` object with `edge_index=[2, num_edges]` convention. My question is,
which one should be passed, `edge_index=[[0, 0, 1], [0, 1, 1]]` or `edge_index=[[0, 1, 1], [0, 0, 1]]` if I use `flow="source_to_target"`

**Comment by wsad1** [comment, marked as answer]
Check out the description of MessagePassing.message(...) [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_gnn.html), based on that to construct a message from A to B, set `edge_index=[[0, 0, 1], [0, 1, 1]]` if `flow='source_to_target'`.

**Reply by zcaicaros** [reply]
Thank you for your confirmation!
